src/services/services.ts

```diff
@@ -969,9 +972,20 @@ module ts {
         getSemanticDiagnostics(fileName: string): Diagnostic[];
         getCompilerOptionsDiagnostics(): Diagnostic[];
 
+        /** 
+         * @deprecated Use getSyntacticClassifications2 instead.
+         */
         getSyntacticClassifications(fileName: string, span: TextSpan): ClassifiedSpan[];
+
+        /** 
+         * @deprecated Use getSemanticClassifications2 instead.
+         */
         getSemanticClassifications(fileName: string, span: TextSpan): ClassifiedSpan[];
 
+        // Encoded as triples of [start, length, ClassificationType].  
+        getSyntacticClassifications2(fileName: string, span: TextSpan): Classifications;

```

Comment:
Noooooooooooooooooooooooooooooo... can we find a better name.. i do not think numbers belong in API names